lib/persistent.js

```diff
@@ -220,28 +220,24 @@ module.exports = class Persistent {
     req.reply(null)
   }
 
-  static signMutable (seq, value, secretKey) {
-    const signature = Buffer.allocUnsafe(64)
+  static signMutable (seq, value, secretKeyOrSigner) {
     const signable = Buffer.allocUnsafe(32)
     sodium.crypto_generichash(signable, c.encode(m.mutableSignable, { seq, value }), NS.MUTABLE_PUT)
-    sodium.crypto_sign_detached(signature, signable, secretKey)
-    return signature
+    return sign(signable, secretKeyOrSigner)
   }
 
   static verifyMutable (signature, seq, value, publicKey) {
     return verifyMutable(signature, seq, value, publicKey)
   }
 
-  static signAnnounce (target, token, id, ann, secretKey) {
-    const signature = Buffer.allocUnsafe(64)
-    sodium.crypto_sign_detached(signature, annSignable(target, token, id, ann, NS.ANNOUNCE), secretKey)
-    return signature
+  static signAnnounce (target, token, id, ann, secretKeyOrSigner) {
+    const signable = annSignable(target, token, id, ann, NS.ANNOUNCE)
+    return sign(signable, secretKeyOrSigner)
   }
 
-  static signUnannounce (target, token, id, ann, secretKey) {
-    const signature = Buffer.allocUnsafe(64)
-    sodium.crypto_sign_detached(signature, annSignable(target, token, id, ann, NS.UNANNOUNCE), secretKey)
-    return signature
+  static signUnannounce (target, token, id, ann, secretKeyOrSigner) {
```

Comment:
instead of having the argument here be polymorphic can we always instead expect it to be a function and then just have a default one?

Reply:
Same for the other ones